src/Symfony/Component/Validator/Constraints/DataUriValidator.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+/**
+ * @author Kev <https://github.com/symfonyaml>
+ *
+ * @see https://datatracker.ietf.org/doc/html/rfc2397
+ */
+class DataUriValidator extends ConstraintValidator
+{
+    /** maximum number of characters allowed in a error message */
+    private const MAX_MESSAGE_VALUE_LENGTH = 30;
+    /** data-uri regexp */
+    public const PATTERN = '~^
+            data:
+            (?:\w+\/(?:(?!;).)+)?  # MIME-type
+            (?:;[\w\W]*?[^;])*     # parameters
+            (;base64)?             # encoding
+            ,
+            [^$]+                  # data
```

Comment:
I don't think this is correct: the data part cannot be "anything-but-$"
Here is another regexp for your consideration:
```
{^data:
(?:([a-zA-Z]++/[-a-zA-Z0-9.+]++)?
(?:;[-a-zA-Z]++=[-a-zA-Z0-9.+]++)*+
(?:;?base64)?
,(?:[-a-zA-Z0-9!$&',()*+;=._~:@/?%\s]*+)
$}
```

Reply:
@nicolas-grekas thank you so much, I'll have a go...

Reply:
Note that I'm not sure my regexp is correct. I often see data URI with html tags in them and this wouldn't be allowed. Looks like carefully reading the RFC my be needed :)